### PR TITLE
Add docs to enums and enum values.

### PR DESCRIPTION
**Pull request by tjprescott**
Required PR due to: https://github.com/Azure/typespec-azure/pull/3006

This will also require a release once merged.

### CODE REVIEW

File: .changeset/modern-worms-exercise.md
Comment (timotheeguerin) [inline .changeset/modern-worms-exercise.md:1]: This should have been the changelog, you can click the comment on the PR to auto populate with the commit message

File: packages/cadl-ranch-specs/http/azure/core/basic/main.tsp
Comment (timotheeguerin) [inline packages/cadl-ranch-specs/http/azure/core/basic/main.tsp:25]: comment here too but do we really need that doc on version enum, not sure what are people going to be commenting apart from the same thing again

File: .changeset/modern-worms-exercise.md
Comment (tjprescott) [reply .changeset/modern-worms-exercise.md:1]: It said to use `--empty` since there is no functional, relevant change to cadl ranch.

File: packages/cadl-ranch-specs/http/azure/core/basic/main.tsp
Comment (tjprescott) [reply packages/cadl-ranch-specs/http/azure/core/basic/main.tsp:25]: This PR is just to make the e2e test pass.

File: .changeset/modern-worms-exercise.md
Comment (timotheeguerin) [reply .changeset/modern-worms-exercise.md:1]: It won’t bum the version when releasing without change

Comment (timotheeguerin) [reply .changeset/modern-worms-exercise.md:1]: make a patch change

File: packages/cadl-ranch-specs/http/azure/core/basic/main.tsp
Comment (tjprescott) [reply packages/cadl-ranch-specs/http/azure/core/basic/main.tsp:25]: The point of the rule is to make sure that enums and enum values have documentation. It's a slippery slope to try and decide that some enums are self-explanatory enough not to require it. If they are, then you could just suppress it.

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/azure/core/basic/main.tsp:25]: But would the version enum be carried over to the generated client?

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/azure/core/basic/main.tsp:25]: sure in general we can't, but for `Versions` we can know it is the version enum. I am saying that as this is a change all the spec will most likely need a change and I don't think is very valuable. But not gonna block on this.

**Comment by tjprescott** [comment]
@timotheeguerin so somehow you and I are both codeowners and you have approved this, but I can't merge it.

**Comment by timotheeguerin** [comment]
hhm, I think this is because DPG teams didn't want change to scenarios merged without approval from them

**Review by lmazuel — APPROVED** [review]

**Pull request merged**
